nvm.sh

```diff
@@ -1858,6 +1874,61 @@ nvm_get_mirror() {
   esac
 }
 
+# args: os, prefixed version, version, tarball, extract directory
+nvm_install_binary_extract() {
+  [ "$#" -ne 5 ] && nvm_err "nvm_install_binary_extract needs 5 parameters" && return 1
+
+  local NVM_OS
+  local PREFIXED_VERSION
+  local VERSION
+  local TARBALL
+  local TMPDIR
+  NVM_OS="${1}"
+  PREFIXED_VERSION="${2}"
+  VERSION="${3}"
+  TARBALL="${4}"
+  TMPDIR="${5}"
+
+  local VERSION_PATH
+
+  [ -n "${TMPDIR-}" ] && \
+  command mkdir -p "${TMPDIR}" && \
+  VERSION_PATH="$(nvm_version_path "${PREFIXED_VERSION}")" || return 1
+
+  # For Windows system (GitBash with MSYS, Cygwin)
+  if [ "${NVM_OS}" = 'win' ]; then
+    VERSION_PATH="${VERSION_PATH}/bin"
+    command unzip -q "${TARBALL}" -d "${TMPDIR}" || return 1
```

Comment:
In order to get this to actually work on Cygwin, I had to add this line right after the unzip:

```sh
command chmod -f +x "${VERSION_PATH}"/{node.exe,npm,npx}
```

The node zipfiles do not have the executable bit set on their contents, so installing without the chmod causes the installation not to be usable (or even recognized by nvm itself as a valid install).  I imagine it would apply to MSYS as well, but I haven't tested.  (The `-f` handles the case where npx is not included on older node versions.)

Reply:
Indeed, I was more using MSYS and everything is executable there.

Reply:
does that mean this line should be added here, for cygwin?

Reply:
Yes.

Reply:
It looks like the chmods were added further down, actually.  See line 1924 below.

Reply:
Awesome. They could still use a comment, explaining it's for cygwin, but LGTM.